ts/main.ts

```diff
@@ -48,17 +52,44 @@ $entryForm.addEventListener('submit', (event: Event) => {
   const title = $titleInput.value;
   const photoUrl = $photoUrlInput.value;
   const notes = $notesTextArea.value;
-  const entryValues: Entry = {
-    title,
-    photoUrl,
-    notes,
-    entryId: data.nextEntryId,
-  };
-  data.nextEntryId++;
-  data.entries.unshift(entryValues);
-  writeData();
-  const $newEntry = renderEntry(entryValues);
-  $allEntriesUl.prepend($newEntry);
+  if (data.editing === null) {
+    const entryValues: Entry = {
+      title,
+      photoUrl,
+      notes,
+      entryId: data.nextEntryId,
+    };
+    data.nextEntryId++;
+    data.entries.unshift(entryValues);
+    writeData();
+    const $newEntry = renderEntry(entryValues);
+    $allEntriesUl.prepend($newEntry);
+  } else {
+    const entryValues: Entry = {
+      title,
+      photoUrl,
+      notes,
+      entryId: data.editing.entryId,
+    };
+    let editEntryIndex = -1;
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === data.editing.entryId) {
+        editEntryIndex = i;
+        break;
+      }
+    }
+    data.entries[editEntryIndex] = entryValues;
+    writeData();
+    const $editedEntry = renderEntry(entryValues);
+    const $replaceEntry = document.querySelector(
+      `li[data-entry-id="${data.editing.entryId}"]`,
+    );
+    if (!$replaceEntry) throw new Error('$replaceEntry query failed');
+    $replaceEntry.replaceWith($editedEntry);
+    $entryFormHeader.textContent = 'New Entry';
+    data.editing = null;
+    writeData();
+  }
```

Comment:
`writeData` is also used in both of your conditions. Let's move this below the conditions since we are writing to local storage in both cases already.